components/main-chef-wrapper/main.go

```diff
@@ -0,0 +1,123 @@
+
+	// We want to pass every sub-command to the old 'chef' CLI binary that was renamed to
+	// 'chef-cli`, which is our default case.
+	default:
+		// When we land in the default case where we run the old 'chef' cli binary,
+		// we need to send the sub-command as well as all the arguments.
+		allArgs = append([]string{subCommand}, allArgs...)
+		cmd = exec.Command("chef-cli", allArgs...)
+	}
+
+	debugLog(fmt.Sprintf("Chef binary: %s\nArguments: %v", cmd.Path, allArgs))
+
+	cmd.Env = os.Environ()
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	// TODO @afiune handle the errors in a better way
+	if err := cmd.Run(); err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			os.Exit(exitError.ExitCode())
+		}
+		// @afiune This might only be needed when doing debugging
+		// because if we got here it means we have a different error
+		// other than a 'ExitError' so print
+		debugLog(fmt.Sprintf("Unexpected Error: \n%s", err))
+		os.Exit(7)
+	}
+}
+
+func usage() {
+	// TODO @afiune add actual usage, this might only list top level sub-commands
+	// we should avoid to add specific options per sub-command
+	fmt.Printf(`Usage:
+    chef -h/--help
+    chef -v/--version
+    chef command [arguments...] [options...]
+
+Available Commands:
+    exec                    Runs the command in context of the embedded ruby
+    env                     Prints environment variables used by Chef Workstation
+    gem                     Runs the 'gem' command in context of the embedded Ruby
+    generate                Generate a new repository, cookbook, or other component
+    shell-init              Initialize your shell to use Chef Workstation as your primary Ruby
+    install                 Install cookbooks from a Policyfile and generate a locked cookbook set
+    update                  Updates a Policyfile.lock.json with latest run_list and cookbooks
+    push                    Push a local policy lock to a policy group on the Chef Infra Server
+    push-archive            Push a policy archive to a policy group on the Chef Infra Server
+    show-policy             Show policyfile objects on the Chef Infra Server
+    diff                    Generate an itemized diff of two Policyfile lock documents
+    export                  Export a policy lock as a Chef Infra Zero code repo
+    clean-policy-revisions  Delete unused policy revisions on the Chef Infra Server
+    clean-policy-cookbooks  Delete unused policyfile cookbooks on the Chef Infra Server
+    delete-policy-group     Delete a policy group on the Chef Infra Server
+    delete-policy           Delete all revisions of a policy on the Chef Infra Server
+    undelete                Undo a delete command
+    describe-cookbook       Prints cookbook checksum information used for cookbook identifier
+
+Experimental Commands:
+    analyze                 Analyze your Chef inventory
+`)
+}
+
+func debugLog(msg string) {
+	if os.Getenv("DEBUG") == "true" {
+		fmt.Println(msg)
```

Comment:
[nit] I usually find it helpful for debug logs to go to stderr.
```suggestion
		fmt.Fprintln(os.Stderr, msg)
```